spatialmedia/gui.py

```diff
@@ -38,75 +39,229 @@
         print("Tkinter library is not available.")
         exit(0)
 
+class Console(object):
+  def __init__(self):
+    self.log = []
+
+  def append(self, text):
+    print text
+    self.log.append(text)
+
+
 
 class Application(Frame):
-    def open(self):
+    def action_open(self):
         """Triggers open file diaglog, reading a new file's metadata."""
-        self.in_file = tkFileDialog.askopenfilename(**self.open_options)
-        if self.in_file:
-            self.set_text("Opened file: %s\n" % self.in_file)
-            spherical.parse_metadata(self.in_file, self.append_text)
+        tmp_in_file = tkFileDialog.askopenfilename(**self.open_options)
+        if not tmp_in_file:
+          return
+        self.in_file = tmp_in_file
+
+        self.set_message("Opened file: %s\n" % ntpath.basename(self.in_file))
+
+        console = Console()
+        metadata = spherical.parse_metadata(self.in_file, console.append)
+
+        for line in console.log:
+          print line
+          if "Error" in line:
+            self.set_error("Failed to load file %s" % ntpath.basename(self.in_file))
+            self.var_spherical.set(0)
+            self.disable_state()
+            self.button_open.configure(state="normal")
+            self.button_quit.configure(state="normal")
+            return
+
+        self.enable_state()
+        self.checkbox_spherical.configure(state="normal")
+
+        if not metadata:
+            self.set_message("No metadata found.")
 
-    def inject(self):
+        if metadata:
+            metadata = metadata.itervalues().next()
+            self.var_spherical.set(1)
+
+            self.set_message("Metadata found in %s\n" % ntpath.basename(self.in_file))
+            if "Spherical" in metadata and metadata["Spherical"] == "true":
```

Comment:
Use dict#get to make this simpler. See: https://docs.python.org/2/library/stdtypes.html#dict.get

ie.
if metadata.get("Spherical", false):

Also extract string literal as a constant or reference it from spherical, which is setting the same value?